### PR TITLE
perf: add password reset field verification

### LINKED ISSUE #1608: 重置密码没有对密码长度进行校验
### 是什么版本出现了此问题？

1.4.16

### 使用的什么数据库？

H2

### 使用的哪种方式部署？

Fat Jar

### 在线站点地址

_No response_

### 发生了什么？

通过 shift+alt+h 方式重置密码无规则限制，一些简单密码如：123456 亦可成功重置。
导致于 admin 后台正常重置密码时，原密码（123456）因无法通过密码复杂度校验而无法重置。

### 相关日志输出

_No response_

### 附加信息

https://bbs.halo.run/d/1981

### PR DESCRIPTION
**Pull request by ruibaby**
添加重置密码接口的字段验证。

Close #1608 

Signed-off-by: Ryan Wang <i@ryanc.cc>

### CODE REVIEW

**Review by guqing — APPROVED** [review]

**Review by JohnNiang — CHANGES_REQUESTED** [review]
Hold on. I'm going to add a controller test against this change.

**Review by JohnNiang — CHANGES_REQUESTED** [review]
Hold on. I'm going to add a controller test against this change.

**Review by guqing — APPROVED** [review]

**Comment by JohnNiang** [comment]
Sorry for the delay. I'm going to review those changes.

File: src/test/java/run/halo/app/controller/AdminControllerTest.java [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+import run.halo.app.service.AdminService;
+import run.halo.app.utils.JsonUtils;
+
+/**
+ * Admin controller test.
+ *
+ * @author guqing
+ * @date 2022-02-11
+ */
+@ExtendWith(MockitoExtension.class)
```
Comment (JohnNiang) [inline src/test/java/run/halo/app/controller/AdminControllerTest.java:37]: If we just test controller only, please refer to https://docs.spring.io/spring-boot/docs/2.5.9/reference/html/features.html#features.testing.spring-boot-applications.spring-mvc-tests

File: src/test/java/run/halo/app/controller/AdminControllerTest.java [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+
+    @NotNull
+    private ResultActions sendResetPasswordCodePerform(String param)
+        throws Exception {
+        return this.mockMvc.perform(post("/api/admin/password/code")
+            .content(param)
+            .contentType(MediaType.APPLICATION_JSON));
+    }
+
+    public static class MockAdminService implements AdminService {
```
Comment (JohnNiang) [inline src/test/java/run/halo/app/controller/AdminControllerTest.java:156]: No need to create a blank AdminService.

**Review by JohnNiang — CHANGES_REQUESTED** [review]

File: src/test/java/run/halo/app/controller/AdminControllerTest.java (hunk 2 above)
Comment (JohnNiang) [reply src/test/java/run/halo/app/controller/AdminControllerTest.java:156]: Please be aware of that this is a test for AdminController, not only for parameter verification.

Comment (JohnNiang) [reply src/test/java/run/halo/app/controller/AdminControllerTest.java:156]: This issue won't block this PR from merging. Anyway, we are supposed to consider optimizing unit-test experience later.

**Review by JohnNiang — APPROVED** [review]

**Pull request merged**